src/components/connection-prompt/bluetooth/BluetoothConnectDialog.svelte

```diff
@@ -36,12 +50,24 @@
     {$t('connectMB.pattern.heading')}
   </DialogHeading>
   <p>{$t('connectMB.pattern.subtitle')}</p>
-  <div class="flex justify-center p-20">
-    <PatternMatrix matrix={$patternMatrixState} onMatrixChange={updateMatrix} />
+  <div class="pt-20 pb-10">
+    <div class="flex justify-center">
+      <PatternMatrix
+        matrix={$patternMatrixState}
+        onMatrixChange={updateMatrix}
+        aria-invalid={attemptedToPairWithInvalidPattern} />
+    </div>
+    <p
+      class="mt-15 text-warning text-center {attemptedToPairWithInvalidPattern
```

Comment:
Following ChakraUI, I think this should have `role="alert"` and only be rendered when the `attemptedToPairWithInvalidPattern` is true. You might have to render an empty div while false to save the space required for the message.

Reply:
Good point, I used invisible/visible instead 😄 https://github.com/microbit-foundation/ml-trainer/pull/175/commits/2affd69c1d677c5d77a91ca273a4ff41dbebc131